odoo/addons/base/models/ir_translation.py

```diff
@@ -922,3 +922,44 @@ def get_web_translations_hash(self, mods, lang):
             'multi_lang': len(self.env['res.lang'].sudo().get_installed()) > 1,
         }
         return hashlib.sha1(json.dumps(translation_cache, sort_keys=True).encode()).hexdigest()
+
+    @api.model
+    def _guess_language_of_terms(self, terms, types=None):
+        """Guess the language of the terms based on the translations stored in the database.
+
+        Note that the search relies on some non indexed fields.
+        :param tuple(str) terms: terms used in the application for which we want to guess the language
+        :param set types: if set, limit the search to the given types (increase performance)
+        :return tuple(str, float):
+            - best_candidate (str): the best language (code) candidate or None if none (ex.: fr_BE)
+            - best_candidate_term_ratio (float): ratio of term found in the database for the best_candidate language
+        """
+        params = dict(terms=tuple([t.lower().strip() for t in terms]), main_lang='en_US')
+        if types:
+            params['types'] = tuple(types)
+        restrict_to_types_clause = ' AND type IN %(types)s ' if types else ''
+        # Count is done outside the sub query to avoid duplicate count for the main language
+        self._cr.execute("""
+        WITH lang_terms AS (
+          SELECT DISTINCT lang, 
+                          TRIM(LOWER(value)) term        
+            FROM ir_translation
+           WHERE TRIM(LOWER(value)) IN %(terms)s""" + restrict_to_types_clause + """
+           UNION
+          SELECT DISTINCT %(main_lang)s lang, 
+                          TRIM(LOWER(src)) term
+            FROM ir_translation
+           WHERE TRIM(LOWER(src)) IN %(terms)s""" + restrict_to_types_clause + """
+        )
+           SELECT lang, 
+                  COUNT(DISTINCT term) uniq_term_count
+             FROM lang_terms
+         GROUP BY lang
+         ORDER BY uniq_term_count DESC
+            LIMIT 1
+        """, params=params)
+        candidates = self._cr.fetchall()
+        if not candidates or not candidates[0][1]:
+            return None, 0.
```

Comment:
Why a final "." ? 0 is ok too I guess ?

Reply:
Actually, it was just to enforce the float type of the documentation but indeed it is equivalent (as a division with integers gives float anyway). I have updated with 0 instead of "0.".